prow/cmd/webhook-server/main.go

```diff
@@ -189,6 +192,12 @@ func main() {
 	interrupts.ListenAndServeTLS(&s, certFile, privKeyFile, 5*time.Second)
 }
 
+func isReady(certFile string, privKeyFile string) func() bool {
```

Comment:
I don't think this readiness check makes sense here.
It doesn't seem like we would ever get the point of serving the readiness endpoint with an empty `certFile` or `privKeyFile`. We would return earlier because `handleSecrets` would return an error. It wouldn't make sense to continue on to `ListenAndServeTLS` if these were not populated so we don't need to signal unreadiness, logging and error and terminating like we do above is sufficient.

I think in this case we can consider the server "ready" after calling `interrupts.ListenAndServeTLS` and the server shouldn't become "unready" after that so we could probably replace `isReady` with `func() bool { return true}` as long as we call `ServeReady` after `ListenAndServeTLS`.
Technically there is still a race here since `interrupts.ListenAndServeTLS` starts listening asynchronously, but I think this will be good enough.